src/crypto/abe/hybrid_crypto/cleartext_header.ts

```diff
@@ -0,0 +1,85 @@
+import { logger } from "../../../utils/logger"
+import { Metadata } from "./metadata"
+import { deserializeList, fromBeBytes } from "../../../utils/utils"
+import { SYMMETRIC_KEY_SIZE } from "./encryption_parameters"
+
+export class ClearTextHeader {
+  private _symmetricKey: Uint8Array
+  private _metadata: Metadata
+
+  /* Getter / Setters */
+  public get symmetricKey(): Uint8Array {
+    return this._symmetricKey
+  }
+  public set symmetricKey(value: Uint8Array) {
+    this._symmetricKey = value
+  }
+  public get metadata(): Metadata {
+    return this._metadata
+  }
+  public set metadata(value: Metadata) {
+    this._metadata = value
+  }
+
+  constructor(symmetricKey: Uint8Array, metadata: Metadata) {
+    this._symmetricKey = symmetricKey
+    this._metadata = metadata
+  }
+
+  /**
+   * Deserialize a clear text header.
+   *
+   * @param cleartextHeader a raw header
+   * @returns the object deserialized
+   */
+  public static parseJson(cleartextHeader: Uint8Array): ClearTextHeader {
```

Comment:
Why `parseJson` takes a `Uint8Array`. Since JSON is a text format, it could be clearer to accept only `string` and leave the decode step outside this function? (it will also removed ambiguity with other parsing function that accept binary format?)

Reply:
That's true. I am going to remove this function (being deprecated since de-serialization changed)